Mathlib/AlgebraicTopology/SimplicialSet/Degenerate.lean

```diff
@@ -122,4 +121,115 @@ lemma isIso_of_nonDegenerate (x : X.nonDegenerate n)
   rw [SimplexCategory.isIso_iff_of_epi]
   exact le_antisymm h (SimplexCategory.len_le_of_epi f)
 
+namespace unique_nonDegenerate
+
+/-!
+Auxiliary definitions and lemmas for the lemmas
+`unique_nonDegenerate₁`, `unique_nonDegenerate₂` and
+`unique_nonDegenerate₃` which assert the uniqueness of the
+decomposition obtained in the lemma `exists_nonDegenerate`.
+-/
+
+section
+
+variable {X} {x : X _⦋n⦌}
+  {m₁ m₂ : ℕ} {f₁ : ⦋n⦌ ⟶ ⦋m₁⦌} (hf₁ : SplitEpi f₁)
+  (y₁ : X.nonDegenerate m₁) (hy₁ : x = X.map f₁.op y₁)
+  (f₂ : ⦋n⦌ ⟶ ⦋m₂⦌) (y₂ : X _⦋m₂⦌) (hy₂ : x = X.map f₂.op y₂)
+
+/-- The composition of a section of `f₁` and `f₂`. It is proven below that it
+is the identity, see `g_eq_id`. -/
+private def g := hf₁.section_ ≫ f₂
+
+variable {f₂ y₁ y₂}
+
+include hf₁ hy₁ hy₂
+
+private lemma map_g_op_y₂ : X.map (g hf₁ f₂).op y₂ = y₁ := by
+  dsimp [g]
+  rw [FunctorToTypes.map_comp_apply, ← hy₂, hy₁, ← FunctorToTypes.map_comp_apply, ← op_comp,
+    SplitEpi.id, op_id, FunctorToTypes.map_id_apply]
+
+private lemma isIso_factorThruImage_g :
+    IsIso (factorThruImage (g hf₁ f₂)) := by
+  have := map_g_op_y₂ hf₁ hy₁ hy₂
+  rw [← image.fac (g hf₁ f₂), op_comp, FunctorToTypes.map_comp_apply] at this
+  exact X.isIso_of_nonDegenerate y₁ (factorThruImage (g hf₁ f₂)) _ this
+
+private lemma mono_g : Mono (g hf₁ f₂) := by
+  have := isIso_factorThruImage_g hf₁ hy₁ hy₂
+  rw [← image.fac (g hf₁ f₂)]
+  infer_instance
+
+private lemma le : m₁ ≤ m₂ := by
+  have := isIso_factorThruImage_g hf₁ hy₁ hy₂
+  exact SimplexCategory.len_le_of_mono
+    (factorThruImage (g hf₁ f₂) ≫ image.ι _)
+
+end
+
+variable {X} in
+private lemma g_eq_id {x : X _⦋n⦌} {m : ℕ} {f₁ : ⦋n⦌ ⟶ ⦋m⦌}
+    {y₁ : X.nonDegenerate m} (hy₁ : x = X.map f₁.op y₁)
+    {f₂ : ⦋n⦌ ⟶ ⦋m⦌} {y₂ : X _⦋m⦌} (hy₂ : x = X.map f₂.op y₂) (hf₁ : SplitEpi f₁) :
+    g hf₁ f₂ = 𝟙 _ := by
+  have := mono_g hf₁ hy₁ hy₂
+  apply SimplexCategory.eq_id_of_mono
+
+end unique_nonDegenerate
+
+section
+
+open unique_nonDegenerate
+
+/-!
+The following lemmas `unique_nonDegenerate₁`, `unique_nonDegenerate₂` and
+`unique_nonDegenerate₃` assert the uniqueness of the decomposition
+obtained in the lemma `exists_nonDegenerate`.
+-/
+
+lemma unique_nonDegenerate₁ (x : X _⦋n⦌) {m₁ m₂ : ℕ}
```

Comment:
How about a slight change of names here? `unique_nonDegenerate₁`, `unique_nonDegenerate₂` and `unique_nonDegenerate₃` are a bit hard to remember which is which. Maybe `unique_nonDegenerate_len` (or `unique_nonDegenerate_dim`), `unique_nonDegenerate_simplex` and `unique_nonDegenerate_section` instead?